cmd/pipeline/list.go

```diff
@@ -0,0 +1,217 @@
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pipeline
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	contextCMD "github.com/okteto/okteto/cmd/context"
+	"github.com/okteto/okteto/cmd/utils"
+	"github.com/okteto/okteto/pkg/constants"
+	oktetoErrors "github.com/okteto/okteto/pkg/errors"
+	"github.com/okteto/okteto/pkg/k8s/configmaps"
+	"github.com/okteto/okteto/pkg/model"
+	"github.com/okteto/okteto/pkg/okteto"
+	"github.com/spf13/cobra"
+	"gopkg.in/yaml.v2"
+	"io"
+	apiv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/validation"
+	"k8s.io/client-go/kubernetes"
+	"os"
+	"strings"
+	"text/tabwriter"
+)
+
+type listFlags struct {
+	labels    []string
+	namespace string
+	output    string
+}
+
+type pipelineListItem struct {
+	Name       string   `json:"name" yaml:"name"`
+	Status     string   `json:"status" yaml:"status"`
+	Repository string   `json:"repository" yaml:"repository"`
+	Branch     string   `json:"branch" yaml:"branch"`
+	Labels     []string `json:"labels" yaml:"labels"`
+}
+
+func list(ctx context.Context) *cobra.Command {
+	flags := &listFlags{}
+
+	cmd := &cobra.Command{
+		Use:   "list",
+		Short: "List all okteto pipelines",
+		Args:  utils.NoArgsAccepted(""),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if flags.namespace == "" {
+				flags.namespace = okteto.Context().Namespace
+			}
+
+			ctxResource := &model.ContextResource{}
+			if err := ctxResource.UpdateNamespace(flags.namespace); err != nil {
+				return err
+			}
+
+			ctxOptions := &contextCMD.ContextOptions{
+				Namespace: ctxResource.Namespace,
+				Show:      false,
+			}
+			if flags.output == "" {
+				ctxOptions.Show = true
+			}
+			if err := contextCMD.NewContextCommand().Run(ctx, ctxOptions); err != nil {
+				return err
+			}
+
+			if !okteto.IsOkteto() {
+				return oktetoErrors.ErrContextIsNotOktetoCluster
+			}
+
+			pc, err := NewCommand()
+			if err != nil {
+				return err
+			}
+			c, _, err := pc.k8sClientProvider.Provide(okteto.Context().Cfg)
+			if err != nil {
+				return fmt.Errorf("failed to load okteto context '%s': %v", okteto.Context().Name, err)
+			}
+
+			return executeListPipelines(cmd.Context(), *flags, configmaps.List, getPipelineListOutput, c, os.Stdout)
+		},
+	}
+
+	cmd.Flags().StringArrayVarP(&flags.labels, "label", "", []string{}, "tag and organize dev environments using labels (multiple --label flags accepted)")
```

Comment:
Should we be able to run it into a certtain cluster?

```suggestion
	cmd.Flags().StringVarP(&flags.context, "context", "c", "", "context where the pipelines are deployed (defaults to the current context)")
	cmd.Flags().StringVarP(&flags.namespace, "namespace", "n", "", "namespace where the pipelines are deployed (defaults to the current namespace)")
```

Reply:
Do we want to? I am not sure, considering other commands require switching context before executing them.

Reply:
is the context flag supported by the spec? 🤔  @jLopezbarb

Reply:
> Do we want to? I am not sure, considering other commands require switching context before executing them.

I think it's the same for this one, we need to load a context given the namespace and the context (default to current one). We can talk about this offline before making a decision, but I think it's the same as the rest of pipeline commands

> is the context flag supported by the spec? 

No, but I find it odd to have that flag on all the commands except for this one.

Reply:
as discussed offline, it makes sense to add, so I've updated the PR reflecting this change!!